rust/arrow/src/array/array_union.rs

```diff
@@ -232,7 +231,7 @@ impl UnionArray {
         assert!(index - self.offset() < self.len());
         if self.is_dense() {
             let valid_slots = match self.data.null_buffer() {
-                Some(b) => bit_util::count_set_bits_offset(b.data(), 0, index),
+                Some(b) => b.count_set_bits_offset(0, index),
```

Comment:
Unrelated to this refactoring, but this code in `UnionArray` did not match my understanding of the [array format documentation][1]. At least I don't see any mention that the offsets array is compressed by omitting null values. Or maybe the documentation needs a better example that includes a null bitmap.

 [1]: https://arrow.apache.org/docs/format/Columnar.html#dense-union

Reply:
Oh, I got it, the union format actually changed and deprecated the use of a null bitmap: https://github.com/apache/arrow/commit/6df862096c796f438c1b6cf054f51e2e2228b368#